### PR TITLE
Use different path for babel-loader cache by default

### PR DESCRIPTION
**Pull request by pustovalov**
Hey,

relative to this issue: https://github.com/rails/webpacker/issues/2530

### CODE REVIEW

**Comment by gauravtiwari** [comment]
@pustovalov Thanks for this but I am wondering if we remove all cache from tmp and move it to `node_modules/.cache` by default except compiler hash that can live in tmp. We can hardcode these paths since they are pretty standard and remove config option from webpacker.yml

**Comment by pustovalov** [comment]
@gauravtiwari updated, WDYT?

**Comment by gauravtiwari** [comment]
Looks great, thank you for updating. One minor thing, could you please put path in one place and use `config.cachePath` so we are not hardcoding at several places but instead using variable like before?

**Comment by pustovalov** [comment]
Friendly ping @gauravtiwari

File: package/config.js [hunk 1]
```diff
@@ -36,5 +36,6 @@ const getPublicPath = () => {
 
 config.publicPath = getPublicPath()
 config.publicPathWithoutCDN = `/${config.public_output_path}/`
+config.cachePath = 'node_modules/.cache/webpacker'
```
Comment (gauravtiwari) [inline package/config.js:39]: Thanks for the changes but I think this is handled internally if we set: `{cacheDirectory: true}` so no need to set this for loaders

https://webpack.js.org/loaders/babel-loader/#options

File: lib/webpacker/configuration.rb [hunk 2]
```diff
@@ -59,6 +59,10 @@ def cache_path
     root_path.join(fetch(:cache_path))
   end
 
+  def node_modules_cache_path
```
Comment (gauravtiwari) [inline lib/webpacker/configuration.rb:62]: We can remove this one?

**Comment by gauravtiwari** [comment]
Sorry @pustovalov I didn't notice this before but I think we can simplify this even further by just setting cacheDirectory option tot true and let the loader do the job. I hope that's okay?

**Comment by pustovalov** [comment]
@gauravtiwari updated, WDYT?

**Comment by gauravtiwari** [comment]
Great, thank you 🙏 . Will merge once passes

**Comment by pustovalov** [comment]
it will failed like in master:
https://travis-ci.org/github/rails/webpacker/jobs/692947115

master specs:
https://travis-ci.org/github/rails/webpacker/jobs/692943219

```
ruby-head - #gemset created /home/travis/.rvm/gems/ruby-head@global
ruby-head - #importing gemset /home/travis/.rvm/gemsets/global.gems.............................
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received
The build has been terminated
```

**Pull request merged**